### PR TITLE
Validation model specs

### PR DESCRIPTION
**Pull request by SiansandaTamara**
## Implemented The Following ##
- Add the following validations:
- For the User model:
- Name must not be blank.
- PostsCounter must be an integer greater than or equal to zero.
- For the Post model:
- Title must not be blank.
- Title must not exceed 250 characters.
- CommentsCounter must be an integer greater than or equal to zero.
- LikesCounter must be an integer greater than or equal to zero.
- nAdd unit specs for all of your models' methods and validations.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class User < ApplicationRecord
   has_many :comments, class_name: 'Comment'
   has_many :likes, class_name: 'Like'
 
+  # validations
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than: -1 }
```
Comment (OLIPLICHE) [inline app/models/user.rb]: Please **PostsCounter must be an integer greater than or equal to zero**. Kindly Refer to the Project requirements

**Review by OLIPLICHE — CHANGES_REQUESTED** [review]
Hi @SiansandaTamara , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Linter is green (the last commit have a green check mark (✅)) 👏
- README file is descriptive 👏
- Commit messages are meaningful 👏 👏
- Git flow is correctly used 👏
- The Test is green 👏 👏

Capture



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- Please Update your ReadMe file, and use the template provided by Microverse. I will make your work more professional. 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb [hunk 2]
```diff
@@ -3,6 +3,10 @@ class User < ApplicationRecord
   has_many :comments, class_name: 'Comment'
   has_many :likes, class_name: 'Like'
 
+  # validations
+  validates :name, presence: true
+  validates :posts_counter, numericality: { only_integer: true, greater_than: 0 }
```
Comment (medaminedev66) [inline app/models/user.rb]: - [ ] Posts_counter must be an integer greater than or equal to zero:
```rb
validates :posts_counter, numerically: { only_integer: true, greater_than_or_equal_to: 0 }
```

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    it 'if author_id is not present' do
+      @comment.author_id = nil
+      expect(@comment).to_not be_valid
+    end
+
+    it 'if post_id is not present' do
+      @comment.post_id = nil
+      expect(@comment).to_not be_valid
+    end
+  end
```
Comment (medaminedev66) [inline spec/models/comment_spec.rb:46]: [OPTIONAL] Following the project requirements, any custom methods in the models' User, Post, Like and Comment should have a corresponding unit test.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @SiansandaTamara 😃,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DeliceLydia — APPROVED** [review]
## Hi @SiansandaTamara, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**